packages/block-library/src/details/edit.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * WordPress dependencies
+ */
+import { InnerBlocks, RichText } from '@wordpress/block-editor';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+export default ( { attributes, className, setAttributes } ) => {
+	const toggleOpen = () => setAttributes( { open: ! attributes.open } );
+
+	/* You may be wondering why we don't just use a <details> element here.
+	 * The problem we are trying to solve is that a <summary> is basically a
+	 * button, and when it has focus, it eats the space key.
+	 *
+	 * That's a problem if you want to use a <RichText> component inside your
+	 * <summary>. Each time you press space, it toggles the rest of the
+	 * <details>, and it doesn't even add a space to your <RichText>.
+	 *
+	 * Then there's the fact that the space exists for a11y reasons. If you
+	 * catch the event and preventDefault() then you've remove the way for
+	 * keyboard users to toggle the <details>.
+	 */
+	return (
+		<div className={ className }>
+			<Button isTertiary className="twistie" onClick={ toggleOpen }>
+				{ attributes.open ? '▼' : '▶' }
+			</Button>
```

Comment:
How about just opening the accordion automatically when it or one of its children is selected? That way we don't have to have a separate button for opening the accordion in the editor, which doesn't match the front-end anyway.

In fact, there's no guarantee a `details` element will be styled using a twistie at all. A theme may want to style it to look completely different. But that becomes a lot harder if there's a twistie element that doesn't exist on the front-end implementation.

Instead, the "twistie" should be applied using CSS styling the same way that the default `summary` element gets it (in Firefox, anyway):

```css
summary { list-style-type: "▸ "; }
```

For the editor style, we'd be targeting the `RichText` `div` below, rather than an actual `summary`.

Notably, Chromium doesn't use `list-style-type` to style the `summary` element, and does things a bit more off-spec, using a pseudo-element (summary::-webkit-details-marker). But since we're styling a fake `details`, that doesn't matter for the editor styles. That's just something that themes have to keep in mind for the front-end styles.

Reply:
I struggled with how to style the twistie (a term I learned working on this block) and still mimic the state of `details`. It should be easier to move into the CSS now that we're making the open/close toggle an explicit control instead of trying to match the state.

Thanks for your feedback, I appreciate it.

Reply:
FYI it looks like [`list-style-type: "▶"` doesn't work in Safari](https://caniuse.com/#feat=mdn-css_properties_list-style-type_string) yet. I confirmed locally and it renders as "•", but I'm going to look at using an image/SVG to keep it in CSS.